pkg/api/file_repository.go

```diff
@@ -36,21 +41,40 @@ func NewFileSystemRepository(dir string) (*FileSystemRepository, error) {
 			}
 
 			definition := repo.parseDefinition(appConfigBody)
-			if err = repo.Add(definition); err != nil {
-				log.WithError(err).Error("Can't add the definition to the repository")
-				return nil, err
+			for _, v := range definition.defs {
+				if err = repo.Add(&v); err != nil {
+					log.WithError(err).Error("Can't add the definition" + v.Name + " to the repository")
+					return nil, err
+				}
 			}
 		}
 	}
 
 	return repo, nil
 }
 
-func (r *FileSystemRepository) parseDefinition(apiDef []byte) *Definition {
-	appConfig := NewDefinition()
-	if err := json.Unmarshal(apiDef, appConfig); err != nil {
+//Unmarshal a single Definition
+func unmarshalDefinition(b []byte, d *Definition) error {
+	if err := json.Unmarshal(b, d); err != nil {
 		log.WithError(err).Error("[RPC] --> Couldn't unmarshal api configuration")
+		return err
 	}
+	return nil
+}
+
+func (r *FileSystemRepository) parseDefinition(apiDef []byte) definitionList {
+	appConfigs := definitionList{}
+
+	//Try unmarshalling as if json is an unnamed Array of multiple definitions
+	if err := json.Unmarshal(apiDef, &appConfigs); err != nil {
+		//Try unmarshalling as if json is a single Definition
+		appConfigs.defs = append(appConfigs.defs, *NewDefinition())
```

Comment:
Just so I understand a bit more, why do you do a dereference here? Instead of just using it as a pointer?

Reply:
Hello, Italo.
appConfigs.defs is a []Definition and is uninitialized by that point.
It boils down to the Unmarshaller implementation (d *Definition).UnmarshalJSON receiver. This method is called whenever we try to parse a Definition(single or multiple per file) and it points to the first item of the field apiConfigs.defs in case of Single or the N-th in case of Multiple Defition. Actually, this first element could be created with default values Definition{} as a placeholder only since the Unmarshaller will call NewDefinition() anyway.

Reply:
I got that part. I was just curious why you do ``

Reply:
Oh, I see. 
So, it's beacause of the Unmarshaller receiver for an array of Definitions: It need to be a slice of Definition(not *Definition)

Thanks